homeassistant/components/generic/camera.py

```diff
@@ -80,9 +109,17 @@ def __init__(self, hass, device_info):
         self._authentication = device_info.get(CONF_AUTHENTICATION)
         self._name = device_info.get(CONF_NAME)
         self._still_image_url = device_info[CONF_STILL_IMAGE_URL]
-        self._stream_source = device_info.get(CONF_STREAM_SOURCE)
+        if not isinstance(
+            self._still_image_url, template_helper.Template
+        ) and self._still_image_url not in [None, ""]:
```

Comment:
```suggestion
        ) and self._still_image_url:
```

```
>>> still_image_url = None
>>> bool(still_image_url)
False
>>> still_image_url = ""
>>> bool(still_image_url)
False
>>> 
```

I don't think we need to guard against 0 here.

Reply:
Agree. Done.